src/main/java/io/gravitee/policy/groovy/model/ScriptableExecutionContext.java

```diff
@@ -0,0 +1,131 @@
+import io.gravitee.gateway.reactive.api.context.GenericResponse;
+import io.gravitee.reporter.api.v4.metric.Metrics;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import lombok.Value;
+
+/**
+ * @author Antoine CORDIER (antoine.cordier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Value
+public class ScriptableExecutionContext implements GenericExecutionContext {
+
+    private static final String CONTEXT_DICTIONARIES_VARIABLE = "dictionaries";
+
+    GenericExecutionContext executionContext;
+
+    @SuppressWarnings({ "unchecked", "unused" })
+    public Map<String, String> getDictonaries() {
+        return (Map<String, String>) executionContext
+            .getTemplateEngine()
+            .getTemplateContext()
+            .lookupVariable(CONTEXT_DICTIONARIES_VARIABLE);
+    }
+
+    @Override
+    public GenericRequest request() {
+        return executionContext.request();
+    }
+
+    @Override
+    public GenericResponse response() {
+        return executionContext.response();
+    }
+
+    @Override
+    public Metrics metrics() {
+        return executionContext.metrics();
+    }
+
+    @Override
+    public <T> T getComponent(Class<T> aClass) {
+        return executionContext.getComponent(aClass);
+    }
+
+    @Override
+    public void setAttribute(String s, Object o) {
+        executionContext.setAttribute(s, o);
+    }
+
+    @Override
+    public void putAttribute(String s, Object o) {
+        executionContext.putAttribute(s, o);
+    }
+
+    @Override
+    public void removeAttribute(String s) {
+        executionContext.removeAttribute(s);
+    }
+
+    @Override
+    public <T> T getAttribute(String s) {
+        return executionContext.getAttribute(s);
+    }
+
+    @Override
+    public <T> List<T> getAttributeAsList(String s) {
+        return executionContext.getAttributeAsList(s);
+    }
+
+    @Override
+    public Set<String> getAttributeNames() {
+        return executionContext.getAttributeNames();
+    }
+
+    @Override
+    public <T> Map<String, T> getAttributes() {
+        return executionContext.getAttributes();
+    }
+
+    @Override
+    public void setInternalAttribute(String s, Object o) {
+        executionContext.setInternalAttribute(s, o);
+    }
+
+    @Override
+    public void putInternalAttribute(String s, Object o) {
+        executionContext.putInternalAttribute(s, o);
+    }
+
+    @Override
+    public void removeInternalAttribute(String s) {
+        executionContext.removeInternalAttribute(s);
+    }
+
+    @Override
+    public <T> T getInternalAttribute(String s) {
+        return executionContext.getInternalAttribute(s);
+    }
+
+    @Override
+    public <T> Map<String, T> getInternalAttributes() {
+        return executionContext.getInternalAttributes();
+    }
+
+    @Override
+    public TemplateEngine getTemplateEngine() {
+        return executionContext.getTemplateEngine();
```

Comment:
The EL template engine should not be accessible.

Reply:
✅